sa/storage-authority.go

```diff
@@ -185,11 +191,22 @@ func (ssa *SQLStorageAuthority) GetLatestValidAuthorization(registrationId int64
 	if err != nil {
 		return
 	}
-	err = ssa.dbMap.SelectOne(&authz, "SELECT id, identifier, registrationID, status, expires, challenges, combinations "+
+	err = ssa.dbMap.SelectOne(&authz, "SELECT id, identifier, registrationID, status, expires, combinations "+
 		"FROM authz "+
 		"WHERE identifier = :identifier AND registrationID = :registrationId AND status = 'valid' "+
 		"ORDER BY expires DESC LIMIT 1",
 		map[string]interface{}{"identifier": string(ident), "registrationId": registrationId})
+	if err != nil {
+		return
+	}
+
+	var challs []core.Challenge
+	_, err = ssa.dbMap.Select(&challs, "SELECT * FROM challenges WHERE authorizationID = :authID", map[string]interface{}{"authID": authz.ID})
```

Comment:
In these cases the second query is dependent on information from the first (the authorization ID).